analyzers/src/SonarAnalyzer.CSharp/Rules/RedundancyInConstructorDestructorDeclaration.cs

```diff
@@ -48,11 +48,17 @@ protected override void Initialize(SonarAnalysisContext context)
         private static void CheckTypesWithPrimaryConstructor(SonarSyntaxNodeReportingContext context)
         {
             var typeDeclaration = (TypeDeclarationSyntax)context.Node;
-            if (typeDeclaration.ParameterList() is { Parameters.Count: 0 } parameterList
+            if (!Excluded(typeDeclaration)
+                && typeDeclaration.ParameterList() is { Parameters.Count: 0 } parameterList
                 && !IsStructWithInitializedFieldOrProperty(typeDeclaration, context.SemanticModel))
             {
                 context.ReportIssue(Rule, parameterList, "primary constructor");
             }
+
+            static bool Excluded(TypeDeclarationSyntax node) =>
```

Comment:
As long as there is only one check in here I would rename this to something like `IsSimplifyingBaseConstructor`.

Reply:
yes my bad - I thought about changing it and then forgot it.

Reply:
A no wait I've changed it. You are commenting on the commit.

Reply:
oh wow sorry 🙈